### PR TITLE
feat(@wdio/browser-runner): allow to define custom hostname for component tests

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

implements #13206

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [x] Back-ported PR at `#13213 `

## Further comments

n/a

### CODE REVIEW

File: packages/wdio-browser-runner/src/index.ts [hunk 1]
```diff
@@ -98,7 +97,8 @@ export default class BrowserRunner extends LocalRunner {
 
         try {
             await server.start()
-            runArgs.args.baseUrl = `http://localhost:${server.config.server?.port}`
+            const hostname = this.#options.hostname || DEFAULT_HOSTNAME
+            runArgs.args.baseUrl = `http://${hostname}:${server.config.server?.port}`
```
Comment (HassHz) [inline packages/wdio-browser-runner/src/index.ts:101]: With this change, will it only start allowing specified `hostnames`, but still not allow specifying `https`? I agree from your comment on the issue that `port` doesn't need to be customizable, but what about `https`?

Comment (christian-bromann) [reply packages/wdio-browser-runner/src/index.ts:101]: @HassHz what do you think of calling the option `host` and have the user include protocol and hostname?

Comment (HassHz) [reply packages/wdio-browser-runner/src/index.ts:101]: I believe a combined `host` would work fine for my usage. Would there be a default certificates?
I wonder if having separate `https` and `hostname` option would be needed for others' use cases though... maybe if for whatever reason they need to provide secure certs? It's something the Vite allows, though.
`https: true` or:
```
https: {
  key: ...,
  cert: ...
}
```

Comment (christian-bromann) [reply packages/wdio-browser-runner/src/index.ts:101]: > Would there be some default self-signed certificates?

No, we can combine it to host as well 👍

**Comment by christian-bromann** [comment]
Failing tests are due to known issues I am working on, I will go ahead and merge to fix them in `main`.

**Pull request merged**